resources/views/platform/components/developer/sets-page.blade.php

```diff
@@ -114,25 +141,54 @@
                     @endif
 
                     @if ($userProgress !== null)
-                        <td>
+                        @if ($game['achievements_published'] == 0)
+                            <td></td>
+                        @else
+                            <td>
+                            <?php
+                                $hardcoreProgressBarWidth = $softcoreProgressBarWidth = 0;
+                                $gameProgress = $userProgress[$game['ID']] ?? null;
+                                $achievementsUnlocked = 0;
+                                if ($gameProgress != null && $game['achievements_published']) {
+                                    $achievementsUnlocked = $gameProgress['achievements_unlocked'];
+                                    $hardcoreProgressBarWidth = sprintf("%01.2f", $gameProgress['achievements_unlocked_hardcore'] * 100 / $game['achievements_published']);
+                                    $softcoreProgressBarWidth = sprintf("%01.2f", ($achievementsUnlocked - $gameProgress['achievements_unlocked_hardcore']) * 100 / $game['achievements_published']);
+                                }
+                            ?>
+                            <div role="progressbar" aria-valuemin="0" aria-valuemax="100"
+                                title="{{ $achievementsUnlocked }} of {{ $game['achievements_published'] }} unlocked"
+                                class="w-full h-1 bg-embed rounded flex">
+                                <div style="width: {{ $hardcoreProgressBarWidth }}%"
+                                        class="bg-[#cc9900] h-full {{ $hardcoreProgressBarWidth > 0 ? 'rounded-l' : '' }}"></div>
+                                <div style="width: {{ $softcoreProgressBarWidth }}%"
+                                        class="bg-[rgb(11,113,193)] h-full {{ $hardcoreProgressBarWidth === 0 ? 'rounded-l' : '' }}"></div>
+                            </div>
+                            </td>
```

Comment:
The amount of PHP here suggests this may be worth breaking out as a child component that lives in a separate Blade file.